bloom-housing/bloom
# feat: improve client-side field validation on Partners listings

This PR addresses metrotranscom/doorway#771


## Description

This PR sets up the Date/Time fields from UIC so that they can forbid non-numerical input. It also added a client-side validation trigger to the Listings field, which I tested a lot so it *shouldn't* add any weird edge cases to how saving a Listing works, but that should definitely have a bit of QA attention.

## How Can This Be Tested/Reviewed?

Add or edit a listing, and try to make sure you can't type letters and also nonsense numbers (like the year 400 or 23 for a month) for Date & Time fields like Application Due Date, Open House Dates, submitted time on Applications, etc.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
